pkgs/development/libraries/libcaca/default.nix

```diff
@@ -13,8 +15,16 @@ stdenv.mkDerivation rec {
 
   outputs = [ "bin" "dev" "out" "man" ];
 
-  propagatedBuildInputs = [ ncurses zlib imlib2 pkgconfig libX11 ]
-   ++ stdenv.lib.optional stdenv.isDarwin libXext;
+  configureFlags = [
+    (if x11Support then "--enable-x11" else "--disable-x11")
+    ];
+
+  NIX_CFLAGS_COMPILE = stdenv.lib.optional (!x11Support) "-DX_DISPLAY_MISSING";
```

Comment:
I didn't check the libcaca docs but wanted to make double check just to be sure... explicitly defining `X_DISPLAY_MISSING` is still required even after passing `--enable-x11` to the configure flags?

Reply:
IIRC the code that checks the define is in `imlib2`. Is there something like a `propagatedCPPFlags`? I guess it would be cleaner if `imlib2-nox` propagated that define.

Reply:
Without digging into that much deeper I'm happy with that answer then. Thanks!